File: TextosJapones/src/main/java/org/jisho/textosJapones/database/dao/SincronizacaoDao.java
package org.jisho.textosJapones.database.dao;

import org.jisho.textosJapones.model.entities.Sincronizacao;
import org.jisho.textosJapones.model.enums.Conexao;
import org.jisho.textosJapones.model.exceptions.ExcessaoBd;

public interface SincronizacaoDao {

	void update(Sincronizacao obj) throws ExcessaoBd;

	Sincronizacao select(Conexao tipo) throws ExcessaoBd;

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/database/dao/SincronizacaoDao.kt
package br.com.fenix.processatexto.database.dao

import br.com.fenix.processatexto.model.entities.processatexto.Sincronizacao
import br.com.fenix.processatexto.model.enums.Conexao
import java.sql.SQLException
import java.util.*

interface SincronizacaoDao {
    @Throws(SQLException::class)
    fun insert(obj: Sincronizacao) : Conexao

    @Throws(SQLException::class)
    fun update(obj: Sincronizacao)

    @Throws(SQLException::class)
    fun delete(obj: Sincronizacao)

    @Throws(SQLException::class)
    fun select(tipo: Conexao): Optional<Sincronizacao>
}